src/EditorFeatures/CSharpTest/Diagnostics/AddInheritdoc/AddInheritdocTests.cs

```diff
@@ -0,0 +1,381 @@
+}");
+        }
+
+        [Theory]
+        [InlineData("public void {|CS1591:OtherMethod|}() { }")]
+        [InlineData("public void {|CS1591:M|}() { }")]
+        [InlineData("public new void {|CS1591:M|}() { }")]
+        public async Task DontOfferOnNotOverridenMethod(string methodDefintion)
+        {
+            await TestMissingAsync(
+            $@"
+/// Some doc.
+public class BaseClass
+{{
+    /// Some doc.
+    public virtual void M() {{ }}
+}}
+/// Some doc.
+public class Derived: BaseClass
+{{
+    {methodDefintion}
+}}");
+        }
+
+        [Fact]
+        public async Task AddMissingInheritdocOnImplicitInterfaceMethod()
+        {
+            await TestAsync(
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    void M();
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    public void {|CS1591:M|}() { }
+}",
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    void M();
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    /// <inheritdoc/>
+    public void M() { }
+}");
+        }
+
+        [Fact]
+        public async Task DontOfferOnExplicitInterfaceMethod()
+        {
+            await TestMissingAsync(
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    void M();
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    void IInterface.M() { }
+}");
+        }
+        [Fact]
+        public async Task AddMissingInheritdocOnOverridenProperty()
+        {
+            await TestAsync(
+            @"
+/// Some doc.
+public class BaseClass
+{
+    /// Some doc.
+    public virtual string P { get; set; }
+}
+/// Some doc.
+public class Derived: BaseClass
+{
+    public override string {|CS1591:P|} { get; set; }
+}",
+            @"
+/// Some doc.
+public class BaseClass
+{
+    /// Some doc.
+    public virtual string P { get; set; }
+}
+/// Some doc.
+public class Derived: BaseClass
+{
+    /// <inheritdoc/>
+    public override string P { get; set; }
+}");
+        }
+
+        [Fact]
+        public async Task AddMissingInheritdocOnImplicitInterfaceProperty()
+        {
+            await TestAsync(
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    string P { get; }
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    public string {|CS1591:P|} { get; }
+}",
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    string P { get; }
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    /// <inheritdoc/>
+    public string P { get; }
+}");
+        }
+
+        [Fact]
+        public async Task AddMissingInheritdocOnImplicitInterfaceEvent()
+        {
+            await TestAsync(
+            @"
+/// Some doc.
+public interface IInterface
+{
+    /// Some doc.
+    event System.Action SomeEvent;
+}
+/// Some doc.
+public class MyClass: IInterface
+{
+    public event System.Action {|CS1591:SomeEvent|};
```

Comment:
💡 Can you add tests showing the behavior for this for each of two cases:

1. The interface only declares `SomeEvent`
2. The interface declares both `SomeEvent` and `SomeOtherEvent`

```cs
public event System.Action SomeEvent, SomeOtherEvent;
```